messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushNotificationHandler.kt

```diff
@@ -204,6 +198,35 @@ internal class CustomerIOPushNotificationHandler(
         notificationManager.notify(requestCode, notification)
     }
 
+    private fun createNotificationChannel(
+        channelId: String?,
+        applicationName: String,
+        appMetaData: Bundle?,
+        notificationManager: NotificationManager
+    ) {
+        // Since android Oreo notification channel is needed.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val cioDefaultChannelName = "$applicationName Notifications"
+            val userProvidedChannelName = appMetaData?.getMetaDataString(CIO_METADATA_DEFAULT_CHANNEL_NAME)
+            val channelName = userProvidedChannelName ?: cioDefaultChannelName
+            val channelImportance = appMetaData?.getInt(CIO_METADATA_DEFAULT_CHANNEL_IMPORTANCE, NotificationManager.IMPORTANCE_DEFAULT)
+                ?: NotificationManager.IMPORTANCE_DEFAULT
+
+            if (userProvidedChannelName != null) {
+                pushLogger.logDeleteCioNotificationChannel(cioDefaultChannelName)
```

Comment:
why do we delete the CIO channel? isn't it harmless to just let is be there? instead of trying to delete an already deleted channel everytime?